src/main/java/org/spongepowered/api/state/State.java

```diff
@@ -69,7 +69,7 @@
      * @param name The state property name
      * @return The state property, if available
      */
-    Optional<StateProperty<?>> statePropertyByName(String name);
+    Optional<StateProperty<?>> findStateProperty(String name);
```

Comment:
Could just be `findProperty()` instead?

There is the additional complication that there are other `stateProperty` methods that also return `Optional`s, but they are more tied into getting a property from a key. It's less of a search than using a name is. Is there a differentiation here, or should all of these methods be `findStateProperty`?

I can see the argument for saying "find" is for use by a free form identifier, lack of find when passing some key like object in, but we should decide this now.